src/protocols/secure_channel/CASESession.cpp

```diff
@@ -744,8 +745,24 @@ CHIP_ERROR CASESession::SendSigmaR3()
 
     ChipLogDetail(SecureChannel, "Sending SigmaR3");
 
+    uint32_t opcredsLen = 0;
+    uint8_t opcreds[kMaxCHIPCertLength];
```

Comment:
same here: please allocate large blocks of memory using ScopedMemoryBuffer (on heap/platofmheap)